### ISSUE TITLE
funeval doesn't play well with ForwardDiff

### ISSUE THREAD

**Issue by isutoris** [QuantEcon/BasisMatrices.jl#40]
Hi, 

I've been trying to use automatic differentiation with some expressions that use funeval(). This doesn't seem to work when a dual number is passed as the argument where the function is to be evaluated. It does however work when dual numbers are passed as basis function coefficients.

Example: (versions: Julia 0.6.0 and master checkout of BasisMatrices)
```julia
using BasisMatrices
using ForwardDiff
fspace = Basis(ChebParams(5, -1.0, 1.0))
f = (b, x) -> funeval(b, fspace, x)
b0 = ones(5)
x0 = 0.0
@show ForwardDiff.gradient(b -> f(b,x0), b0)
@show ForwardDiff.derivative(x -> f(b0,x), x0)
```

The last line returns an error:

```
ERROR: MethodError: Cannot `convert` an object of type ForwardDiff.Dual{ForwardDiff.Tag{##5#6,0xb046287d533b082d},Float64,1} to an object of type Float64                                               
This may have arisen from a call to the constructor Float64(...),                                   
since type constructors fall back to convert methods.
```

I understand one can compute derivatives explicitly by passing an order argument to funeval, but in my case things would be much simpler if I could just use automatic differentiation.

**Comment by sglyon** [comment]
Hi @isutoris thanks for reporting and having such a clean example for me to use when looking at the problem.

I think this has been fixed in #41. Can you pull the `sl/eltype_BasisMatrix` branch and test it out?

**Comment by isutoris** [comment]
Wow, that was quick. Yes, the error is gone when using the new branch. Thanks!
